src/plugins/mer/merbuildconfiguration.cpp

```diff
@@ -0,0 +1,129 @@
+** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** Other Usage
+**
+** Alternatively, this file may be used in accordance with the terms and
+** conditions contained in a signed written agreement between you and Digia.
+**
+****************************************************************************/
+
+#include "merbuildconfiguration.h"
+
+#include "merconnection.h"
+#include "merconstants.h"
+#include "merbuildsteps.h"
+
+#include <projectexplorer/buildmanager.h>
+#include <projectexplorer/buildsteplist.h>
+#include <projectexplorer/projectexplorer.h>
+#include <qmakeprojectmanager/qmakeproject.h>
+#include <qmakeprojectmanager/qmakestep.h>
+
+#include <QTextStream>
+#include <QRegularExpression>
+
+using namespace ProjectExplorer;
+using namespace QmakeProjectManager;
+
+namespace {
+
+const char MER_VARIABLES_CACHE_FILENAME[] = ".mb2/qmake_variables.cache";
+
+QString parseVariableString(const QString &line)
+{
+    QRegExp rx("\\s*=\\s*");
+    const int index = line.indexOf(rx);
+    if (index == -1)
+        return QString();
+
+    const QString key = line.mid(0, index);
+    const QString value = line.mid(index + rx.matchedLength());
+
+    return QString("%1=%2").arg(key).arg(value);
+}
+
+QStringList readVariables(const QString &fileName)
+{
+    QStringList result;
+    QFile file(fileName);
+    if (file.exists() && file.open(QIODevice::ReadOnly | QIODevice::Text)) {
+        QTextStream in(&file);
+        while (!in.atEnd()) {
+            const QString line = parseVariableString(in.readLine());
+            if (line.size())
+                result << line;
+        }
+    }
+    return result;
+}
+
+} // namespace anonymous
+
+namespace Mer {
+namespace Internal {
+
+MerBuildConfiguration::MerBuildConfiguration(Target *target, Core::Id id)
+    : QmakeBuildConfiguration(target, id)
+    , m_useExtraParserArguments(false)
+{
+    QmakeProject *qmakeProject = static_cast<QmakeProject *>(project());
+    connect(qmakeProject, &QmakeProject::buildDirectoryInitialized,
+            this, &MerBuildConfiguration::setupExtraParserArguments);
+}
+
+void MerBuildConfiguration::setUseExtraParserArguments(bool value)
+{
+    m_useExtraParserArguments = value;
+    setupExtraParserArguments();
+}
+
+void MerBuildConfiguration::updateExtraParserArguments() const
+{
+    BuildStepList *steps = stepList(Core::Id(ProjectExplorer::Constants::BUILDSTEPS_BUILD));
+    QTC_ASSERT(steps, return);
+    const auto sdkStartStep = qobject_cast<Mer::Internal::MerSdkStartStep *>(steps->at(0));
+    QTC_ASSERT(sdkStartStep, return);
+    if (!sdkStartStep->connection()
+            || sdkStartStep->connection()->state() != Mer::Internal::MerConnection::Connected) {
+        BuildManager::appendStep(sdkStartStep, sdkStartStep->displayName());
+    }
+
+    QMakeStep *qs = qmakeStep();
+    QTC_ASSERT(qs, return);
+    qs->setForced(true);
+    BuildManager::appendStep(qs, tr("Updating cache"));
```

Comment:
This code forces to run qmake step. The use case is:
1. User edit variables in yaml or spec file.
2. Then we need to update cache file.
3. Force to run qmake step and mb2 updates cache.

Reply:
Ah I see, I did't realize it would run it immediately. But then, is it correct to modify the step with setForced here? It's not a temporary instance.

Reply:
As a side not here: we should consider adding an option to `mb2 qmake` so that we could only populate the variable cache and not actually run qmake, and use it here.

Reply:
I think you may be right. setForced seems not needed here (maybe this line leaked here from my old source code).

Reply:
Removed setForced.